tfx/orchestration/portable/beam_executor_operator.py

```diff
@@ -31,6 +33,33 @@
   _BeamPipeline = Any
 
 
+def _resolve_beam_args_from_env(beam_pipeline_args, beam_pipeline_args_from_env) -> list:
+  resolved_beam_pipeline_args_from_env = []
+
+  for beam_pipeline_arg_from_env, env_var in beam_pipeline_args_from_env.items():
+    # If an arg is already present in beam_pipeline_args, it should take precedence
+    # over env vars.
+    if any("--{}=".format(beam_pipeline_arg_from_env) in beam_pipeline_arg for beam_pipeline_arg in beam_pipeline_args):
+      logging.info('Arg %s already present in '
+        'beam_pipeline_args and will not be fetched from env.',
+         beam_pipeline_arg_from_env)
+      continue
+
+    env_var_value = os.getenv(env_var)
+    if env_var_value:
+      if beam_pipeline_arg_from_env.startswith('--'):
+        resolved_beam_pipeline_args_from_env.append('{}={}'
+                  .format(beam_pipeline_arg_from_env, env_var_value))
+      else:
+        resolved_beam_pipeline_args_from_env.append('--{}={}'
+                                                      .format(beam_pipeline_arg_from_env, env_var_value))
+    else:
+      # TODO: Raise value error instead?
```

Comment:
I think that we should handle None and ''(empty string) separately. Because empty string is a valid environment variable value. I'm a bit inclining to raising an exception for missing env var.

Reply:
I agree. I feel that it's generally better to fail fast rather than to obscurely log an issue. Added test case for missing env var.